src/main/java/seedu/duke/AnimeStorage.java

```diff
@@ -0,0 +1,133 @@
+        // For each pathname in the pathnames array
+        //for (String pathname : pathnames) {
+        //    System.out.println(pathname);
+        //}
+    }
+
+    private String prepareFile(String fileFolder) {
+        return RELATIVE_DIR + fileFolder.replace("\\",FILE_SEPARATOR).replace("/",FILE_SEPARATOR);
+    }
+
+    public ArrayList<Anime> readAnimeDatabase() throws IOException {
+        ArrayList<Anime> animeDataList = new ArrayList<>();
+        for (String pathname : pathnames) {
+            FileReader fileData = new FileReader(dataFile.getPath() + FILE_SEPARATOR
+                    + pathname);
+            //System.out.println(fileData);
+            parseJson(animeDataList, fileData);
+        }
+        return animeDataList;
+    }
+
+    private void parseJson(ArrayList<Anime> animeDataList,FileReader  fileData) {
+        JSONParser parser = new JSONParser();
+        JSONArray jsonList = new JSONArray();
+        try {
+            jsonList = (JSONArray) parser.parse(fileData);
+        } catch (ParseException | IOException e) {
+            e.printStackTrace();
+        }
+
+        Iterator<JSONObject> iterator = jsonList.iterator();
+        JSONObject jsonObject = new JSONObject();
+        while (iterator.hasNext()) {
+            String animeName;
+            int animeEpisode = 0;
+            int animeRating = 0;
+            int animeDuration = 0;
+
+            jsonObject = (JSONObject) iterator.next().get("data");
+            jsonObject = (JSONObject) jsonObject.get("Media");
+
+            //Getting anime name
+            JSONObject jsonTitle = (JSONObject) jsonObject.get("title");
+            if (jsonTitle.get("english") == null) {
+                animeName = (String) jsonTitle.get("romaji");
+            } else {
+                animeName = (String) jsonTitle.get("english");
+            }
+
+            //getting anime episode
+
+            if (jsonObject.get("episodes") != null) {
+                animeEpisode =  (int) (long) jsonObject.get("episodes");
+            }
+
+            //getting start date
+            JSONObject jsonDate = (JSONObject) jsonObject.get("startDate");
+            String[] animeReleaseDate;
+            animeReleaseDate = new String[] { String.valueOf((long) jsonDate.get("year")),
+                    String.valueOf((long) jsonDate.get("month")), String.valueOf((long) jsonDate.get("day"))};
+
+            //getting rating
+            if (jsonObject.get("averageScore") != null) {
+                animeRating =  (int) (long) jsonObject.get("averageScore");
+            }
+
+            //getting genre
+            ArrayList<String> animeGenre;
+            animeGenre = new ArrayList<>();
+            JSONArray jsonGenre = (JSONArray) jsonObject.get("genres");
+            Iterator<String> iteratorGenre = jsonGenre.iterator();
+            while (iteratorGenre.hasNext()) {
+                animeGenre.add(iteratorGenre.next());
+            }
```

Comment:
Following the above comment to remove the warning. This chunk of code can be replaced with:
            for (Object genre : jsonGenre) {
                animeGenre.add((String) genre);
            }